sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSession.java

```diff
@@ -223,7 +226,8 @@ public Mono<Void> rollbackTransaction(AmqpTransaction transaction) {
     @Override
     public Mono<AmqpLink> createProducer(String linkName, String entityPath, Duration timeout, AmqpRetryPolicy retry) {
         return createProducer(linkName, entityPath, timeout, retry, null)
-            .or(onClosedError("Connection closed while waiting for new producer link.", entityPath, linkName));
+            .or(onClosedError("Connection closed while waiting for new producer link.", entityPath, linkName))
+            .cast(AmqpLink.class);
```

Comment:
Why is this required?

Reply:
Just keep the same pattern with `createConsumer`, the older code cast to amqplink before this method, I put the cast from overload to this method, same with `createConsumer` did.